packages/test/test-end-to-end-tests/src/test/localTestSignals.spec.ts

```diff
@@ -191,3 +227,101 @@ describeCompat("TestSignals", "FullCompat", (getTestObjectProvider) => {
 		);
 	});
 });
+
+describeCompat("Targeted Signals", "NoCompat", (getTestObjectProvider) => {
+	const numberOfClients = 3;
+	assert(numberOfClients >= 2, "Need at least 2 clients for targeted signals");
+	let clients: SignalClient[];
+	let provider: ITestObjectProvider;
+
+	beforeEach("setup containers", async () => {
+		provider = getTestObjectProvider();
+		clients = [];
+		for (let i = 0; i < numberOfClients; i++) {
+			const container = await (i === 0
+				? provider.makeTestContainer(testContainerConfig)
+				: provider.loadTestContainer(testContainerConfig));
+			const dataObject = await getContainerEntryPointBackCompat<ITestFluidObject>(container);
+
+			if (container.connectionState !== ConnectionState.Connected) {
+				await new Promise((resolve) => container.once("connected", resolve));
+			}
+
+			clients.push({
+				dataStoreRuntime: dataObject.runtime,
+				containerRuntime: dataObject.context.containerRuntime,
+				signalReceivedCount: 0,
+				clientId: container.clientId,
+			});
+		}
+	});
+	async function sendAndVerifyRemoteSignals(runtime: "containerRuntime" | "dataStoreRuntime") {
+		clients.forEach((client) => {
+			client[runtime].on("signal", (message: IInboundSignalMessage, local: boolean) => {
+				assert.equal(local, false, "Signal should be remote");
+				assert.equal(message.type, "TestSignal", "Signal type should be TestSignal");
+				assert.equal(message.content, true, "Signal content should be true");
+				client.signalReceivedCount += 1;
+			});
+		});
+
+		for (let i = 0; i < clients.length; i++) {
+			const targetClient = clients[(i + 1) % clients.length];
+			clients[i][runtime].submitSignal("TestSignal", true, targetClient.clientId);
+			await waitForTargetedSignal(
+				targetClient[runtime],
+				clients.filter((c) => c !== targetClient).map((c) => c[runtime]),
+			);
+		}
+
+		clients.forEach((client, index) => {
+			assert.equal(
+				client.signalReceivedCount,
+				1,
+				`client ${index + 1} did not receive signal`,
+			);
+		});
+	}
+	async function sendAndVerifyLocalSignals(runtime: "containerRuntime" | "dataStoreRuntime") {
+		clients.forEach((client) => {
+			client[runtime].on("signal", (message: IInboundSignalMessage, local: boolean) => {
+				assert.equal(local, true, "Signal should be local");
+				assert.equal(message.type, "TestSignal", "Signal type should be TestSignal");
+				assert.equal(message.content, true, "Signal content should be true");
+				client.signalReceivedCount += 1;
+			});
+		});
+
+		for (const client of clients) {
+			client[runtime].submitSignal("TestSignal", true, client.clientId);
+			await waitForTargetedSignal(
+				client[runtime],
+				clients.filter((c) => c !== client).map((c) => c[runtime]),
+			);
+		}
+
+		clients.forEach((client, index) => {
+			assert.equal(
+				client.signalReceivedCount,
+				1,
+				`client ${index + 1} did not receive signal`,
+			);
+		});
+	}
+
+	it("Validates data store runtime remote signals", async () => {
```

Comment:
I don't know what this is actually testing. What action is performed? What results are checked?

Reply:
I renamed the tests to specify that that the tests are for targeting a specific remote or local client using their client ID

Reply:
Maybe these tests involve too many things to be captured in simple phrases and a larger descriptive comment is needed.

Reply:
Just added larger descriptive comments to help fully explain what is being tested in the helper functions.